core/trino-main/src/main/java/io/trino/sql/planner/LogicalPlanner.java

```diff
@@ -451,29 +452,36 @@ private RelationPlan getInsertPlan(
                 continue;
             }
             Symbol output = symbolAllocator.newSymbol(column.getName(), column.getType());
+            Expression expression;
+            Type tableType = column.getType();
             int index = insertColumns.indexOf(columns.get(column.getName()));
             if (index < 0) {
                 if (supportsMissingColumnsOnInsert) {
                     continue;
                 }
-                Expression cast = new Cast(new NullLiteral(), toSqlType(column.getType()));
-                assignments.put(output, cast);
-                insertedColumnsBuilder.add(column);
+                if (!column.isNullable()) {
+                    expression = createNullNotAllowedFailExpression(column.getName(), tableType);
```

Comment:
Does `createNullNotAllowedFailExpression` get added twice?

Reply:
Depends how you interpret adding twice.
If this one is executed than we end up with `coalesce(createNullNotAllowedFailExpression, createNullNotAllowedFailExpression)`.
Throwing exception from here broke a lot of tests and made it really hard to have a common behaviour between different connectors

Reply:
> If this one is executed than we end up with `coalesce(createNullNotAllowedFailExpression, createNullNotAllowedFailExpression)`.

That's redundant, and easy to avoid.

Reply:
Why not just

```
expression =  new Cast(new NullLiteral(), toSqlType(column.getType()));
```

?